### ISSUE TITLE
Travis: ccache not working properly

### ISSUE THREAD

**Issue by MonsieurNicolas** [stellar/stellar-core#2160]
Looks like Travis' cache is not working as expected.

For example, looking at the *compilation leg*
https://www.travis-ci.org/stellar/stellar-core/jobs/547531450

the cache contains ~400MB of data at the end of the build:
```
ccache -s
cache directory                     /home/travis/.ccache
primary config                      /home/travis/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                   175
cache hit (preprocessed)             125
cache miss                           804
called for link                        8
called for preprocessing              82
compile failed                        12
preprocessor error                     8
bad compiler arguments                18
unsupported source language            4
autoconf compile/link                210
no input file                         59
files in cache                      1205
cache size                         416.0 MB
max cache size                     500.0 MB
```

but right after, the *test leg* ( https://www.travis-ci.org/stellar/stellar-core/jobs/547531452 ) doesn't seem to take advantage of it:

```
real	20m10.691s
user	38m3.702s
sys	1m55.214s
ccache -s
cache directory                     /home/travis/.ccache
primary config                      /home/travis/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                   289
cache hit (preprocessed)             247
cache miss                          1081
called for link                       11
called for preprocessing             112
compile failed                        17
preprocessor error                    12
bad compiler arguments                23
unsupported source language            6
autoconf compile/link                305
no input file                         74
files in cache                      1269
cache size                         417.8 MB
max cache size                     500.0 MB
```

Basically it has even more cache misses than the "build only" leg!?

According to Travis cache stats, the cache size is only 142MB for the entire `auto` branch (I would expect it to be close to 800MB as we have `clang` and `gcc` flavors)

**Comment by MonsieurNicolas** [comment]
Well... I tried https://github.com/stellar/stellar-core/pull/2162
and it looks like things are better now, the output of ccache still doesn't make sense though (given the number of cache misses)
```
real	0m53.236s
user	1m20.141s
sys	0m16.856s
ccache -s
cache directory                     /home/travis/.ccache
primary config                      /home/travis/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                  1003
cache hit (preprocessed)             525
cache miss                          2157
called for link                      108
called for preprocessing             211
compile failed                        42
preprocessor error                    21
bad compiler arguments                35
unsupported source language           14
autoconf compile/link                686
no input file                        107
files in cache                      3487
cache size                         483.4 MB
max cache size                     900.0 MB
```

**Comment by MonsieurNicolas** [comment]
fixed by #2162
